### DISCUSSION TITLE
Beat map loading

### DISCUSSION THREAD

**Discussion by Emp1r1c** [WebOsu-2/webosu-2.github.io#18] [Ideas]
Since most people play Web Osu in school and around half of the beat map loaders don't work why don't we make it so you can choose what beat map loader you want or need at that time? the one issue is if you like a song but it does not work on the other loader and some songs don't work just have them not download (tbh I would not know how to do that but its just a idea)

**Comment by Mostlime12195** [comment]
I completely get that, and I tried to do that with Mino and SayoBot a while ago. The reason I was not able to do it is that it would require a refactor of the code (and I frankly don't know enough about it to do such a thing) since Mino and SayoBot give different formats of data.

My solution was to instead have two sets of code--one for Mino, only executed when it is selected, and the other for SayoBot--but that very quickly spiraled out of control with a MASSIVE amount of code that made everything messy and unreadable. I had to undo everything because I couldn't stand it.
